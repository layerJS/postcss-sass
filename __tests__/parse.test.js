const jsonify = require('postcss-parser-tests').jsonify;
const scssSyntax = require('postcss-scss');
const path = require('path');
const fs = require('fs');

const parse = require('../parse');
let tests = fs
    .readdirSync(path.join(__dirname, 'cases'))
    .filter(i => path.extname(i) === '.sass' );

function read(file) {
    return fs.readFileSync(path.join(__dirname, 'cases', file)).toString();
}

for ( let name of tests ) {
    it('parses ' + name, () => {
        const sass = read(name);
        const scss = read(name.replace(/\.\w+$/, '.scss'));
        const root = parse(sass, {
            from: path.posix.join('__tests__/cases', name)
        });
        // fs.writeFile(path.posix.join('__tests__/cases', name.replace(/\.\w+$/, '.scss')), root.toString(scssSyntax));
        expect(root.toString(scssSyntax)).toEqual(scss);
        expect(jsonify(root)).toMatchSnapshot();
    });
}
