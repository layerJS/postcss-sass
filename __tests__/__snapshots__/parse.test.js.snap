// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses comments.sass 1`] = `
"{
  \\"raws\\": {
    \\"before\\": \\"\\"
  },
  \\"type\\": \\"root\\",
  \\"nodes\\": [
    {
      \\"raws\\": {
        \\"before\\": \\"\\",
        \\"inline\\": false,
        \\"left\\": \\" \\",
        \\"right\\": \\" \\",
        \\"text\\": \\"multi\\\\n   line\\"
      },
      \\"type\\": \\"comment\\",
      \\"text\\": \\"multi\\\\n   line\\"
    },
    {
      \\"raws\\": {
        \\"before\\": \\"\\\\n\\",
        \\"inline\\": true,
        \\"left\\": \\" \\",
        \\"right\\": \\"\\",
        \\"text\\": \\"inline\\"
      },
      \\"type\\": \\"comment\\",
      \\"text\\": \\"inline\\"
    },
    {
      \\"raws\\": {
        \\"before\\": \\"\\\\n\\",
        \\"between\\": \\"\\"
      },
      \\"type\\": \\"rule\\",
      \\"nodes\\": [
        {
          \\"raws\\": {
            \\"before\\": \\"\\\\n  \\",
            \\"between\\": \\": \\",
            \\"semicolon\\": false
          },
          \\"type\\": \\"decl\\",
          \\"prop\\": \\"color\\",
          \\"value\\": \\"black\\",
          \\"source\\": {
            \\"start\\": {
              \\"line\\": 5,
              \\"column\\": 3
            },
            \\"end\\": {
              \\"line\\": 5,
              \\"column\\": 14
            },
            \\"input\\": {
              \\"file\\": \\"comments.sass\\"
            }
          }
        },
        {
          \\"raws\\": {
            \\"before\\": \\"\\\\n \\",
            \\"inline\\": true,
            \\"left\\": \\" \\",
            \\"right\\": \\"\\",
            \\"text\\": \\"last\\"
          },
          \\"type\\": \\"comment\\",
          \\"text\\": \\"last\\"
        },
        {
          \\"raws\\": {
            \\"before\\": \\"\\\\n  \\",
            \\"between\\": \\": \\",
            \\"semicolon\\": false
          },
          \\"type\\": \\"decl\\",
          \\"prop\\": \\"z-index\\",
          \\"value\\": \\"1\\",
          \\"source\\": {
            \\"start\\": {
              \\"line\\": 6,
              \\"column\\": 3
            },
            \\"end\\": {
              \\"line\\": 6,
              \\"column\\": 12
            },
            \\"input\\": {
              \\"file\\": \\"comments.sass\\"
            }
          }
        }
      ],
      \\"selector\\": \\"a\\",
      \\"source\\": {
        \\"start\\": {
          \\"line\\": 4,
          \\"column\\": 1
        },
        \\"end\\": {
          \\"line\\": 6,
          \\"column\\": 13
        },
        \\"input\\": {
          \\"file\\": \\"comments.sass\\"
        }
      }
    },
    {
      \\"raws\\": {
        \\"before\\": \\"\\\\n\\\\n\\",
        \\"inline\\": true,
        \\"left\\": \\" \\",
        \\"right\\": \\"\\",
        \\"text\\": \\"a\\\\n  // color: black\\"
      },
      \\"type\\": \\"comment\\",
      \\"text\\": \\"a\\\\n  // color: black\\"
    }
  ],
  \\"source\\": {
    \\"start\\": {
      \\"line\\": 1,
      \\"column\\": 1
    },
    \\"end\\": {
      \\"line\\": 9,
      \\"column\\": 18
    },
    \\"input\\": {
      \\"file\\": \\"comments.sass\\"
    }
  }
}"
`;

exports[`parses empty.sass 1`] = `
"{
  \\"raws\\": {
    \\"before\\": \\"\\"
  },
  \\"type\\": \\"root\\",
  \\"nodes\\": [
    {
      \\"raws\\": {
        \\"before\\": \\"\\",
        \\"inline\\": true,
        \\"left\\": \\"\\",
        \\"right\\": \\"\\"
      },
      \\"type\\": \\"comment\\",
      \\"text\\": \\"\\"
    }
  ],
  \\"source\\": {
    \\"start\\": {
      \\"line\\": 1,
      \\"column\\": 1
    },
    \\"end\\": {
      \\"line\\": 1,
      \\"column\\": 3
    },
    \\"input\\": {
      \\"file\\": \\"empty.sass\\"
    }
  }
}"
`;

exports[`parses important.sass 1`] = `
"{
  \\"raws\\": {
    \\"before\\": \\"\\"
  },
  \\"type\\": \\"root\\",
  \\"nodes\\": [
    {
      \\"raws\\": {
        \\"before\\": \\"\\",
        \\"between\\": \\": \\",
        \\"semicolon\\": false,
        \\"important\\": \\" !important\\"
      },
      \\"type\\": \\"decl\\",
      \\"prop\\": \\"long\\",
      \\"important\\": true,
      \\"value\\": \\"black\\",
      \\"source\\": {
        \\"start\\": {
          \\"line\\": 1,
          \\"column\\": 1
        },
        \\"end\\": {
          \\"line\\": 1,
          \\"column\\": 22
        },
        \\"input\\": {
          \\"file\\": \\"important.sass\\"
        }
      }
    },
    {
      \\"raws\\": {
        \\"before\\": \\"\\\\n\\",
        \\"between\\": \\": \\",
        \\"semicolon\\": false,
        \\"important\\": \\"!important\\"
      },
      \\"type\\": \\"decl\\",
      \\"prop\\": \\"short\\",
      \\"important\\": true,
      \\"value\\": \\"black\\",
      \\"source\\": {
        \\"start\\": {
          \\"line\\": 2,
          \\"column\\": 1
        },
        \\"end\\": {
          \\"line\\": 2,
          \\"column\\": 22
        },
        \\"input\\": {
          \\"file\\": \\"important.sass\\"
        }
      }
    }
  ],
  \\"source\\": {
    \\"start\\": {
      \\"line\\": 1,
      \\"column\\": 1
    },
    \\"end\\": {
      \\"line\\": 2,
      \\"column\\": 23
    },
    \\"input\\": {
      \\"file\\": \\"important.sass\\"
    }
  }
}"
`;
